pretyping/evarconv.ml

```diff
@@ -695,6 +702,7 @@ let rec evar_conv_x flags env evd pbty term1 term2 =
         end
 
 and evar_eqappr_x ?(rhs_is_already_stuck = false) flags env evd pbty
+    parents1 parents2 keys1 keys2 lastUnfolded
```

Comment:
IMO, it would make more sense to define a `type cs_keys_cache = { .... }` and pass a single argument here.
The type should come with an `empty : unit -> ..` constructor and a `flip` function to exchange left with right.
It would also help to document what the last argument (the last unfolded thing) is.